docs/misc/config.md

```diff
@@ -0,0 +1,131 @@
+Configuration
+=============
+
+Configurations define settings and consumer preferences that can be applied to a mod instance. Forge uses a configuration system using [TOML][toml] files and read with [NightConfig][nightconfig].
+
+Creating a Configuration
+------------------------
+
+A configuration can be created using a subtype of `IConfigSpec`. Forge implements the type via `ForgeConfigSpec` and enables its construction through `ForgeConfigSpec$Builder`. The builder can separate the config values into sections via `Builder#push` to create a section and `Builder#pop` to leave a section. Afterwards, the configuration can be built using one of two methods:
+
+Method    | Description
+:---      | :---
+`build`     | Creates the `ForgeConfigSpec`.
+`configure` | Creates a pair of the class holding the config values and the `ForgeConfigSpec`.
+
+!!! note
+    `ForgeConfigSpec$Builder#configure` is typically used with a `static` block and a class that takes in `ForgeCondifSpec$Builder` as part of its constructor to attach and hold the values:
+
+    ```java
+    // In some config class
+    ExampleConfig(ForgeConfigSpec.Builder builder) {
+        // Define values here in final fields
+    }
+
+    // Somewhere the constructor is accessible
+    static {
+        Pair<ExampleConfig, ForgeConfigSpec> pair = new ForgeConfigSpec.Builder()
+            .configure(ExampleConfig::new);
+        // Store pair values in some constant field
+    }
+    ```
+
+Each config value can be supplied with additional context to provide additional behavior. Contexts must be defined before the config value is fully built:
+
+Method       | Description
+:---         | :---
+`comment`      | Provides a description of what the config value does. Can provide multiple strings for a multiline comment.
+`translation`  | Provides a translation key for the name of the config value.
+`worldRestart` | The world must be restarted before the config value can be changed.
+
+### ConfigValue
+
+Config values can be built with the provided contexts (if defined) using any of the `#define` methods.
+
+All config value methods take in at least two components:
+* A path representing the name of the variable: a `.` separated string representing the sections the config value is in
+* The default value when no valid configuration is present
+
+The `ConfigValue` specific methods take in two additional components:
+* A validator to make sure the deserialized object is valid
+* A class representing the data type of the config value
```

Comment:
The absence of a space between the paragraph and the list is causing the list to not render properly.
![image](https://user-images.githubusercontent.com/21304337/161408730-a44e4f5a-a7f8-4a62-b39c-c6c37be18cbb.png)
```suggestion
All config value methods take in at least two components:

* A path representing the name of the variable: a `.` separated string representing the sections the config value is in
* The default value when no valid configuration is present

The `ConfigValue` specific methods take in two additional components:

* A validator to make sure the deserialized object is valid
* A class representing the data type of the config value
```